gitlab/resource_gitlab_project_mirror_test.go

```diff
@@ -0,0 +1,210 @@
+			{
+				Config: testAccGitlabProjectMirrorConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectMirrorExists("gitlab_project_mirror.foo", &hook),
+					testAccCheckGitlabProjectMirrorAttributes(&hook, &testAccGitlabProjectMirrorExpectedAttributes{
+						URL:                   fmt.Sprintf("https://example.com/hook-%d", rInt),
+						Enabled:               true,
+						OnlyProtectedBranches: true,
+						KeepDivergentRefs:     true,
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGitlabProjectMirror_import(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectMirrorDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGitlabProjectMirrorConfig(rInt),
+			},
+			{
+				ResourceName:      "gitlab_project_mirror.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabProjectMirrorExists(n string, mirror *gitlab.ProjectMirror) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		splitID := strings.Split(rs.Primary.ID, ":")
+
+		mirrorID, err := strconv.Atoi(splitID[len(splitID)-1])
+		if err != nil {
+			return err
+		}
+		repoName := rs.Primary.Attributes["project"]
+		if repoName == "" {
+			return fmt.Errorf("No project ID is set")
+		}
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		mirrors, _, err := conn.ProjectMirrors.ListProjectMirror(repoName, nil)
+		if err != nil {
+			return err
+		}
+
+		for _, m := range mirrors {
+			if m.ID == mirrorID {
+				*mirror = *m
+				break
+			}
+			return errors.New("unable to find mirror.")
+		}
+		return nil
+	}
+}
+
+type testAccGitlabProjectMirrorExpectedAttributes struct {
+	URL                   string
+	Enabled               bool
+	OnlyProtectedBranches bool
+	KeepDivergentRefs     bool
+}
+
+func testAccCheckGitlabProjectMirrorAttributes(mirror *gitlab.ProjectMirror, want *testAccGitlabProjectMirrorExpectedAttributes) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if mirror.URL != want.URL {
+			return fmt.Errorf("got url %q; want %q", mirror.URL, want.URL)
+		}
+
+		if mirror.Enabled != want.Enabled {
+			return fmt.Errorf("got enabled %t; want %t", mirror.Enabled, want.Enabled)
+		}
+		if mirror.OnlyProtectedBranches != want.OnlyProtectedBranches {
+			return fmt.Errorf("got only protected branches %t; want %t", mirror.OnlyProtectedBranches, want.OnlyProtectedBranches)
+		}
+		if mirror.KeepDivergentRefs != want.KeepDivergentRefs {
+			return fmt.Errorf("got keep divergent refs %t; want %t", mirror.KeepDivergentRefs, want.KeepDivergentRefs)
+		}
+		return nil
+	}
+}
+
+func testAccCheckGitlabProjectMirrorDestroy(s *terraform.State) error {
```

Comment:
Just an fyi, a "destroy" test should be testing the behavior of your own resource destroy function, i.e. checking that the mirror gets disabled. There's an open issue about this generally #205 so we can come back and do this later.

Reply:
sure, I will not make any changes in this PR for this.